### ISSUE TITLE
update_by_query/10_basic/wait_for_completion=false fails

### ISSUE THREAD

**Issue by javanna** [elastic/elasticsearch#19541]
`update_by_query/10_basic/wait_for_completion=false fails` failed quite a few times recently in our 2.4 branch.

Last failure is here: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+2.4+multijob-os-compatibility/os=sles/43/console .

Seems like the tasks.get call that waits for completion gets back a node failure, which indicates that the node timed out:

```
Suite: org.elasticsearch.index.reindex.ReindexRestIT
  2> REPRODUCE WITH: mvn verify -Pdev -Dskip.unit.tests -pl org.elasticsearch.module:reindex -Dtests.seed=76760A523E13F0FB -Dtests.class=org.elasticsearch.index.reindex.ReindexRestIT -Dtests.method="test {yaml=update_by_query/10_basic/wait_for_completion=false}" -Des.logger.level=ERROR -Dtests.assertion.disabled=false -Dtests.security.manager=true -Dtests.heap.size=512m -Dtests.cluster=localhost:9500 -Dtests.rest.suite=reindex,update_by_query -Dtests.locale=el -Dtests.timezone=America/Fortaleza
FAILURE 30.2s | ReindexRestIT.test {yaml=update_by_query/10_basic/wait_for_completion=false} <<<
   > Throwable #1: java.lang.AssertionError: field [node_failures] has a true value but it shouldn't
   > Expected: ("" or equalToIgnoringCase("false") or "0")
   >      but: was "[{type=failed_node_exception, reason=Failed node [BNvZfzA9SdCXzLhGhiIclA], caused_by={type=timeout_exception, reason=Timed out waiting for completion of [org.elasticsearch.index.reindex.BulkByScrollTask@265f94f8]}}]"
   > 	at __randomizedtesting.SeedInfo.seed([76760A523E13F0FB:FE22358890EF9D03]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.test.rest.section.IsFalseAssertion.doAssert(IsFalseAssertion.java:51)
   > 	at org.elasticsearch.test.rest.section.Assertion.execute(Assertion.java:69)
   > 	at org.elasticsearch.test.rest.ESRestTestCase.test(ESRestTestCase.java:375)
   > 	at java.lang.Thread.run(Thread.java:745)
```

**Comment by javanna** [comment]
hey @nik9000 I think you did #19453 while hunting this. The last failure has more info that might interest you.

**Comment by nik9000** [comment]
Indeed. Thanks for pointing me to it!

**Comment by nik9000** [comment]
Luckily I've been able to get this to fail locally with:
```
while mvn verify -Dskip.unit.tests -pl modules/reindex/ -Pdev -Dtests.seed=76760A523E13F0FB -Des.logger.level=info -Dtests.output=always; do echo not yet; done
```

**Comment by nik9000** [comment]
Not every iteration, but after a few minutes.

**Comment by nik9000** [comment]
There is a race condition in 2.x when you rethrottle tasks. Usually speeding up a task will take effect immediately. If you hit it at the right time then speeding up the task will wait until the next batch. This isn't a big deal in real life because you can resend the rethrottle command if you hit the race condition. You probably won't hit it anyway because you'd have to send it in a fairly small window. And it wouldn't matter that much for normal throttle values because you just wait an extra batch.

But in testing we schedule the task to finish 3000 years in the future. And we hit the race condition relatively frequently because we rethrottle immediately after sending the request.

In master this was fixed by 5b94c4a25b8a6322de573d5d9923d5838db748ec in master. Backporting that isn't simple because the implementations have drifted too far apart. So instead I basically silenced the test:
01bf02e56aad670430e0ee808f47eaffad2d0a13 - improve logging
4e2971710dce7a0648f15804bb4ddab8cd5ef09b - tighten the assertions and explain a race condition

**Comment by nik9000** [comment]
I'm going to look at forward porting 01bf02e for nicer logging in master. I'll open a PR about it.
